datadog_checks_base/datadog_checks/base/utils/db/utils.py

```diff
@@ -182,19 +182,22 @@ def default_json_event_encoding(o):
 
 
 def obfuscate_sql_with_metadata(query, options=None):
-    try:
-        statement = datadog_agent.obfuscate_sql(query, options)
-        statement_with_metadata = json.loads(statement)
-    except ValueError:  # Backwards compatible exception between py38 and py27 for json.loads()
-        # Assume we're running against an older agent and return the obfuscated query without metadata.
-        return {'query': statement, 'metadata': {}}
-    except Exception as e:
-        raise e
-    metadata = statement_with_metadata.get('metadata', {})
-    tables = metadata.pop('tables_csv', None)
-    tables = [table.strip() for table in tables.split(',') if table != ''] if tables else None
-    statement_with_metadata['metadata']['tables'] = tables
-    return statement_with_metadata
+    def _load_metadata(statement):
+        try:
+            statement_with_metadata = json.loads(statement)
+            metadata = statement_with_metadata.get('metadata', {})
+            tables = metadata.pop('tables_csv', None)
+            tables = [table.strip() for table in tables.split(',') if table != ''] if tables else None
+            statement_with_metadata['metadata']['tables'] = tables
+            return statement_with_metadata
+        except ValueError:
```

Comment:
You're not catching other Exceptions anymore?

Reply:
Yeah, with the shift in logic, this part here: https://github.com/DataDog/integrations-core/pull/11038/commits/b37664652d4f18c009300d5d124bd604c8faa4ae
Didn't make much sense anymore because catching just to raise again doesn't really do much here because there is no logic underneath (unlike the previous iteration).